src/WidgetBase.ts

```diff
@@ -385,6 +391,13 @@ export class WidgetBase<P extends WidgetProperties> extends Evented implements W
 			const { children, properties = {} } = dNode;
 			const { key } = properties;
 
+			const propagatedPropertyNames = this.getDecorator('propagateProperty');
```

Comment:
I was thinking more that there would be be a an method that would take care of the propagation.

```ts
@afterRender
propagateProperties(result: DNode) {
    const propagatedPropertyNames = this.getDecorator('propagateProperty');
    decorate(result, (dnode: DNode) => { /* add properties */ }, isWNode);
}
```

And looking at this makes me think that it could really be a separate mixin that added this `afterRender`.

Perhaps I'm way off base here.